scripts/build-on-docker.sh

```diff
@@ -1,36 +1,47 @@
 #!/bin/bash
-
-# Travis currently only provides two cores per VM:
-# https://docs.travis-ci.com/user/reference/overview/.
-# There is at least one open issue asking that this info be provided to the
-# client environment: https://github.com/travis-ci/travis-ci/issues/4696.
-# For now just hardcode the expected value.
-CORE_COUNT=2
-
 set -e
 export LANG=en_US.UTF-8
-export CFLAGS='-fno-strict-aliasing -Wno-unknown-pragmas -Wno-missing-braces -Wno-unused-result -Wno-return-type -Wno-int-to-pointer-cast -Wno-parentheses -Wno-unused -Wno-unused-but-set-variable -Wno-cpp -Wno-char-subscripts'
+export CC=gcc  # see the "compiler: gcc" line in the .travis.yml config file
+export CFLAGS="-fno-strict-aliasing -Wno-unknown-pragmas -Wno-missing-braces"
```

Comment:
This code is used in macOS build script too. We should move code that is common to both scripts in a different file and source it.

Reply:
Will do.